website/content/docs/discovery/dns.mdx

```diff
@@ -96,6 +96,41 @@ pairs according to [RFC1464](https://www.ietf.org/rfc/rfc1464.txt).
 Alternatively, the TXT record will only include the node's metadata value when the
 node's metadata key starts with `rfc1035-`.
 
+## Alternative Domain
+
+By default, Consul responds to DNS queries only for its configured
+[`domain`](/docs/agent/options#domain).
+
+Some use cases require responding to queries for more than one domain,
+such as during a DNS migration or to distinguish between internal and
+external queries by using different domains.
+
+Consul can be configured to respond to DNS queries on an alternative domain
+through the [`alt_domain`](/docs/agent/options#alt_domain) agent configuration
+option. Consul's DNS response will use the same domain as was used in the query.
```

Comment:
I just realized we don't indicate which version(s) this statement is true for. @kbabuadze, @trujillo-adam : what do you think of this?

```suggestion
option. As of Consul versions 1.11.0+, Consul's DNS response will use the same domain as was used in the query;
in prior versions, the response may use the primary [`domain`](/docs/agent/options#domain) no matter which
domain was used in the query.
```

Reply:
@trujillo-adam : an idea for you to consider longer term (outside the scope of this PR) ... should sections like this have a version compatibility table? Something like the below... It could allow us to be more explicit about where support starts and how it changes (e.g., bugfixes, enhancements).

|------------------|------------------------|
| version range   | support                      |
| <1.5.2               | not available              |
| 1.5.2 - 1.10       | Consul DNS will respond to queries on the `alt_domain`, but the response may use the primary domain (e.g., `consul.`)
| 1.11.0+            | Consul DNS will respond to queries on the `alt_domain` with the same domain as was used in the query |